src/components/message-actions.js

```diff
@@ -43,9 +43,24 @@ class MessageActions extends CustomElement {
         ev.preventDefault();
         this.chatview.onMessageRetractButtonClicked(this.model);
     }
+    
+    onUserMuteButtonClicked (ev) {
+        ev.preventDefault();
+        this.chatview.onUserMuteButtonClicked(this.model);
```

Comment:
Here I am following the same pattern as the other functions above it. Also I like the idea of having components contain minimal functionality and the maximum level of abstraction possible, for separation of concerns, letting them handle the view and putting side effects elsewhere. Is that unreasonable in any way?

Reply:
We're in the process of getting rid of Backbone/Skeletor Views, which is why I don't want us to put new methods into them if it can be avoided.

We *are* practicing separation of concerns, by putting side effects in the headless part. Looking at the `onMuteUser` method you wrote in `converse-chatview.js`, almost all that code an go into `converse-chat.js`.